testnet/stacks-node/src/config.rs

```diff
@@ -833,7 +834,12 @@ impl Config {
         }
 
         let miner = match config_file.miner {
-            Some(miner) => miner.into_config_default(miner_default_config)?,
+            Some(mut miner) => {
+                if miner.mining_key.is_none() && !node.seed.is_empty() {
+                    miner.mining_key = Some(to_hex(&node.seed));
```

Comment:
AFAIK `node.seed` is just a random string of bytes (a `Vec<u8>`). Are you sure that any valid value for `node.seed` will result in a valid `Secp256k1PrivateKey`?

Reply:
It looks like `node.seed` is supposed to be a valid private key, even though it's not checked. From `mining.md`:

```toml
# Bitcoin private key to spend
seed = "YOUR PRIVATE KEY"
```

And from what I'm reading, any 256-bit integer between `1` and `0xfffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141` is a valid Secp256k1 private key, so there's a small chance of `Secp256k1PrivateKey::from_hex()` failing and the node failing to start (if someone uses `0` for `node.seed`, or about 1 in 2^128 if using the RNG), which I think is acceptable